### ISSUE TITLE
Publish

### ISSUE THREAD

**Issue by EagleAiDi** [a-mos/InteractiveMatting#2]
Hello author, thank you for this magnificent piece of work. It has been greatly enlightening to me. However, I haven't been able to find this paper in conferences or journals. Have you considered publishing it in a journal or at a conference? This would allow for a better citation of your excellent work Thanks!

**Comment by a-mos** [comment]
> Hello author, thank you for this magnificent piece of work. It has been greatly enlightening to me. However, I haven't been able to find this paper in conferences or journals. Have you considered publishing it in a journal or at a conference? This would allow for a better citation of your excellent work Thanks!

Hello, thank you for your interest! This work is just proof of concept and was [presented as an abstract](https://lomonosov-msu.ru/archive/Lomonosov_2023/data/28734/uid512361_report.pdf) at a local conference:

```
@inproceedings{amoskalenko,
  title={Development of a neural network method for interactive image matting},
  author={Andrey Moskalenko},
  booktitle={International scientific conference for undergraduate and graduate students and young scientists "Lomonosov 2023"},
  year={2023},
}
```

**Comment by EagleAiDi** [comment]
> > Hello author, thank you for this magnificent piece of work. It has been greatly enlightening to me. However, I haven't been able to find this paper in conferences or journals. Have you considered publishing it in a journal or at a conference? This would allow for a better citation of your excellent work Thanks!
> 
> Hello, thank you for your interest! This work is just proof of concept and was [presented as an abstract](https://lomonosov-msu.ru/archive/Lomonosov_2023/data/28734/uid512361_report.pdf) at a local conference:
> 
> ```
> @inproceedings{amoskalenko,
>   title={Development of a neural network method for interactive image matting},
>   author={Andrey Moskalenko},
>   booktitle={International scientific conference for undergraduate and graduate students and young scientists "Lomonosov 2023"},
>   year={2023},
> }
> ```
Thank you thank you very much ！

**Comment by EagleAiDi** [comment]
Hi, sorry to bother you again, I read the literature you gave, and tested some of my own images with the weights you gave, its performance is very good, but I have a few questions to consult  you.                                   
1、the paper mentioned the use of Deeplabv3 as the basic architecture, but I did not find this part in the code;only in the training and testing saw Resnet50, the final is achieved with Resnet50?                                
2,、you give the weight to test some of my own images, the effect is very good, in the paper, there is mentioned:Дляобучения использовалась конкатенация из нескольких доступных да-тасетов с эталонными картами прозрачности Для обучения исполь-зовалось 80% от каждого датасета, для валидации оставшиеся 20%. ,May I ask what weights are trained on which data sets?  Thanks!

**Comment by a-mos** [comment]
Hello and thank you for your interest! 

1) The model used is an encoder-decoder network inspired by DeepLab with a ResNet50 encoder, ASPP pooling and simple decoder - https://github.com/a-mos/InteractiveMatting/blob/main/training/model.py#L106
2) I tested 3 setups for model training as shown on https://github.com/a-mos/InteractiveMatting/blob/main/image.png

2.1) Orange line model was trained only with Compositions1K training set with 95/5 train-val split (this model shows worst quality on test benchmark)

The blue line model was trained on following concatenation of datasets: 
```
PhotoMatte85 = StaticTransformDataset(path.join(static_root, 'PhotoMatte85'), method=1, bg_path=os.path.join(static_root, 'Backgrounds'))
AIM500 = StaticTransformDataset(path.join(static_root, 'AIM500'), method=2)
AM2000 = StaticTransformDataset(path.join(static_root, 'AM2000'), method=2)
Distinctions = StaticTransformDataset(path.join(static_root, 'Distinctions'), method=3, bg_path=os.path.join(static_root, 'Backgrounds'))
AISEGMENT = StaticTransformDataset(path.join(static_root, 'aisegmentcom-matting-human-datasets/combined'), method=0)
static_dataset = ConcatDataset([PhotoMatte85, AIM500, AM2000, Distinctions, AISEGMENT])
```
2.2) This concatenated dataset mostly consists of human photos (due to the size of aisegment part) and mentioned as 'Train on human' in figure with blue and green lines. For this pretraining, 80/20 split was used for validation. I think it would be better performance if replace aisegment with recently proposed P3M dataset since aisegment has not really accurate alphas. 

2.3) Best (green) model with attached checkpoint (https://drive.google.com/file/d/1MiTQ1RG7AClm-4VqASXnYTeOImgOTxGm/view) was obtained by additional finetuning of blue model on 95% of Compotions1K training set (and 5% remains for validation).

**Comment by a-mos** [comment]
> Hi, sorry to bother you again, I read the literature you gave, and tested some of my own images with the weights you gave, its performance is very good, but I have a few questions to consult you. 1、the paper mentioned the use of Deeplabv3 as the basic architecture, but I did not find this part in the code;only in the training and testing saw Resnet50, the final is achieved with Resnet50? 2,、you give the weight to test some of my own images, the effect is very good, in the paper, there is mentioned:Дляобучения использовалась конкатенация из нескольких доступных да-тасетов с эталонными картами прозрачности Для обучения исполь-зовалось 80% от каждого датасета, для валидации оставшиеся 20%. ,May I ask what weights are trained on which data sets? Thanks!

Hello and thank you for your interest! 

1) The model used is an encoder-decoder network inspired by DeepLab with a ResNet50 encoder, ASPP pooling and simple decoder - https://github.com/a-mos/InteractiveMatting/blob/main/training/model.py#L106
2) I tested 3 setups for model training as shown on https://github.com/a-mos/InteractiveMatting/blob/main/image.png

2.1) Orange line model was trained only with Compositions1K training set with 95/5 train-val split (this model shows worst quality on test benchmark)

The blue line model was trained on following concatenation of datasets: 
```
PhotoMatte85 = StaticTransformDataset(path.join(static_root, 'PhotoMatte85'), method=1, bg_path=os.path.join(static_root, 'Backgrounds'))
AIM500 = StaticTransformDataset(path.join(static_root, 'AIM500'), method=2)
AM2000 = StaticTransformDataset(path.join(static_root, 'AM2000'), method=2)
Distinctions = StaticTransformDataset(path.join(static_root, 'Distinctions'), method=3, bg_path=os.path.join(static_root, 'Backgrounds'))
AISEGMENT = StaticTransformDataset(path.join(static_root, 'aisegmentcom-matting-human-datasets/combined'), method=0)
static_dataset = ConcatDataset([PhotoMatte85, AIM500, AM2000, Distinctions, AISEGMENT])
```
Backgrouds came from https://grail.cs.washington.edu/projects/background-matting-v2/#/datasets
2.2) This concatenated dataset mostly consists of human photos (due to the size of aisegment part) and mentioned as 'Train on human' in figure with blue and green lines. For this pretraining, 80/20 split was used for validation. I think it would be better performance if replace aisegment with recently proposed P3M dataset since aisegment has not really accurate alphas. 

2.3) Best (green) model with attached checkpoint (https://drive.google.com/file/d/1MiTQ1RG7AClm-4VqASXnYTeOImgOTxGm/view) was obtained by additional finetuning of blue model on 95% of Compotions1K training set (and 5% remains for validation).

**Comment by EagleAiDi** [comment]
Thank you very much for your patient and specific reply on above request ! I will try to re-train .

**Comment by EagleAiDi** [comment]
> > Hi, sorry to bother you again, I read the literature you gave, and tested some of my own images with the weights you gave, its performance is very good, but I have a few questions to consult you. 1、the paper mentioned the use of Deeplabv3 as the basic architecture, but I did not find this part in the code;only in the training and testing saw Resnet50, the final is achieved with Resnet50? 2,、you give the weight to test some of my own images, the effect is very good, in the paper, there is mentioned:Дляобучения использовалась конкатенация из нескольких доступных да-тасетов с эталонными картами прозрачности Для обучения исполь-зовалось 80% от каждого датасета, для валидации оставшиеся 20%. ,May I ask what weights are trained on which data sets? Thanks!
> 
> Hello and thank you for your interest!
> 
> 1. The model used is an encoder-decoder network inspired by DeepLab with a ResNet50 encoder, ASPP pooling and simple decoder - https://github.com/a-mos/InteractiveMatting/blob/main/training/model.py#L106
> 2. I tested 3 setups for model training as shown on https://github.com/a-mos/InteractiveMatting/blob/main/image.png
> 
> 2.1) Orange line model was trained only with Compositions1K training set with 95/5 train-val split (this model shows worst quality on test benchmark)
> 
> The blue line model was trained on following concatenation of datasets:
> 
> ```
> PhotoMatte85 = StaticTransformDataset(path.join(static_root, 'PhotoMatte85'), method=1, bg_path=os.path.join(static_root, 'Backgrounds'))
> AIM500 = StaticTransformDataset(path.join(static_root, 'AIM500'), method=2)
> AM2000 = StaticTransformDataset(path.join(static_root, 'AM2000'), method=2)
> Distinctions = StaticTransformDataset(path.join(static_root, 'Distinctions'), method=3, bg_path=os.path.join(static_root, 'Backgrounds'))
> AISEGMENT = StaticTransformDataset(path.join(static_root, 'aisegmentcom-matting-human-datasets/combined'), method=0)
> static_dataset = ConcatDataset([PhotoMatte85, AIM500, AM2000, Distinctions, AISEGMENT])
> ```
> 
> Backgrouds came from https://grail.cs.washington.edu/projects/background-matting-v2/#/datasets 2.2) This concatenated dataset mostly consists of human photos (due to the size of aisegment part) and mentioned as 'Train on human' in figure with blue and green lines. For this pretraining, 80/20 split was used for validation. I think it would be better performance if replace aisegment with recently proposed P3M dataset since aisegment has not really accurate alphas.
> 
> 2.3) Best (green) model with attached checkpoint (https://drive.google.com/file/d/1MiTQ1RG7AClm-4VqASXnYTeOImgOTxGm/view) was obtained by additional finetuning of blue model on 95% of Compotions1K training set (and 5% remains for validation).

Thank you very much for your patience and the detailed reply to my previous request!

**Comment by EagleAiDi** [comment]
> > Hi, sorry to bother you again, I read the literature you gave, and tested some of my own images with the weights you gave, its performance is very good, but I have a few questions to consult you. 1、the paper mentioned the use of Deeplabv3 as the basic architecture, but I did not find this part in the code;only in the training and testing saw Resnet50, the final is achieved with Resnet50? 2,、you give the weight to test some of my own images, the effect is very good, in the paper, there is mentioned:Дляобучения использовалась конкатенация из нескольких доступных да-тасетов с эталонными картами прозрачности Для обучения исполь-зовалось 80% от каждого датасета, для валидации оставшиеся 20%. ,May I ask what weights are trained on which data sets? Thanks!
> 
> Hello and thank you for your interest!
> 
> 1. The model used is an encoder-decoder network inspired by DeepLab with a ResNet50 encoder, ASPP pooling and simple decoder - https://github.com/a-mos/InteractiveMatting/blob/main/training/model.py#L106
> 2. I tested 3 setups for model training as shown on https://github.com/a-mos/InteractiveMatting/blob/main/image.png
> 
> 2.1) Orange line model was trained only with Compositions1K training set with 95/5 train-val split (this model shows worst quality on test benchmark)
> 
> The blue line model was trained on following concatenation of datasets:
> 
> ```
> PhotoMatte85 = StaticTransformDataset(path.join(static_root, 'PhotoMatte85'), method=1, bg_path=os.path.join(static_root, 'Backgrounds'))
> AIM500 = StaticTransformDataset(path.join(static_root, 'AIM500'), method=2)
> AM2000 = StaticTransformDataset(path.join(static_root, 'AM2000'), method=2)
> Distinctions = StaticTransformDataset(path.join(static_root, 'Distinctions'), method=3, bg_path=os.path.join(static_root, 'Backgrounds'))
> AISEGMENT = StaticTransformDataset(path.join(static_root, 'aisegmentcom-matting-human-datasets/combined'), method=0)
> static_dataset = ConcatDataset([PhotoMatte85, AIM500, AM2000, Distinctions, AISEGMENT])
> ```
> 
> Backgrouds came from https://grail.cs.washington.edu/projects/background-matting-v2/#/datasets 2.2) This concatenated dataset mostly consists of human photos (due to the size of aisegment part) and mentioned as 'Train on human' in figure with blue and green lines. For this pretraining, 80/20 split was used for validation. I think it would be better performance if replace aisegment with recently proposed P3M dataset since aisegment has not really accurate alphas.
> 
> 2.3) Best (green) model with attached checkpoint (https://drive.google.com/file/d/1MiTQ1RG7AClm-4VqASXnYTeOImgOTxGm/view) was obtained by additional finetuning of blue model on 95% of Compotions1K training set (and 5% remains for validation).

Thank you very much for your patience and specific reply  on above request!

**Comment by EagleAiDi** [comment]
> > Hi, sorry to bother you again, I read the literature you gave, and tested some of my own images with the weights you gave, its performance is very good, but I have a few questions to consult you. 1、the paper mentioned the use of Deeplabv3 as the basic architecture, but I did not find this part in the code;only in the training and testing saw Resnet50, the final is achieved with Resnet50? 2,、you give the weight to test some of my own images, the effect is very good, in the paper, there is mentioned:Дляобучения использовалась конкатенация из нескольких доступных да-тасетов с эталонными картами прозрачности Для обучения исполь-зовалось 80% от каждого датасета, для валидации оставшиеся 20%. ,May I ask what weights are trained on which data sets? Thanks!
> 
> Hello and thank you for your interest!
> 
> 1. The model used is an encoder-decoder network inspired by DeepLab with a ResNet50 encoder, ASPP pooling and simple decoder - https://github.com/a-mos/InteractiveMatting/blob/main/training/model.py#L106
> 2. I tested 3 setups for model training as shown on https://github.com/a-mos/InteractiveMatting/blob/main/image.png
> 
> 2.1) Orange line model was trained only with Compositions1K training set with 95/5 train-val split (this model shows worst quality on test benchmark)
> 
> The blue line model was trained on following concatenation of datasets:
> 
> ```
> PhotoMatte85 = StaticTransformDataset(path.join(static_root, 'PhotoMatte85'), method=1, bg_path=os.path.join(static_root, 'Backgrounds'))
> AIM500 = StaticTransformDataset(path.join(static_root, 'AIM500'), method=2)
> AM2000 = StaticTransformDataset(path.join(static_root, 'AM2000'), method=2)
> Distinctions = StaticTransformDataset(path.join(static_root, 'Distinctions'), method=3, bg_path=os.path.join(static_root, 'Backgrounds'))
> AISEGMENT = StaticTransformDataset(path.join(static_root, 'aisegmentcom-matting-human-datasets/combined'), method=0)
> static_dataset = ConcatDataset([PhotoMatte85, AIM500, AM2000, Distinctions, AISEGMENT])
> ```
> 
> Backgrouds came from https://grail.cs.washington.edu/projects/background-matting-v2/#/datasets 2.2) This concatenated dataset mostly consists of human photos (due to the size of aisegment part) and mentioned as 'Train on human' in figure with blue and green lines. For this pretraining, 80/20 split was used for validation. I think it would be better performance if replace aisegment with recently proposed P3M dataset since aisegment has not really accurate alphas.
> 
> 2.3) Best (green) model with attached checkpoint (https://drive.google.com/file/d/1MiTQ1RG7AClm-4VqASXnYTeOImgOTxGm/view) was obtained by additional finetuning of blue model on 95% of Compotions1K training set (and 5% remains for validation).

Thank you very much for your patience and specific reply on above request! Could you please explain if the fine-tuning was applied only to the last few layers of the decoder, or did it involve the entire model?

**Comment by a-mos** [comment]
> Thank you very much for your patience and specific reply on above request! Could you please explain if the fine-tuning was applied only to the last few layers of the decoder, or did it involve the entire model?

Fine-tuning updates the full model, but with 3x lower learning rate (1e-4) than initial training (3e-4).

**Comment by EagleAiDi** [comment]
> > Thank you very much for your patience and specific reply on above request! Could you please explain if the fine-tuning was applied only to the last few layers of the decoder, or did it involve the entire model?
> 
> Fine-tuning updates the full model, but with 3x lower learning rate (1e-4) than initial training (3e-4).

Thank you for your reply, it is very helpful to me.

**Comment by EagleAiDi** [comment]
> > Thank you very much for your patience and specific reply on above request! Could you please explain if the fine-tuning was applied only to the last few layers of the decoder, or did it involve the entire model?
> 
> Fine-tuning updates the full model, but with 3x lower learning rate (1e-4) than initial training (3e-4).

Thank you for your reply again, it is very helpful to me! Thanks!
